exporters/metric/cortex/cortex_test.go

```diff
@@ -54,3 +61,180 @@ func TestInstallNewPipeline(t *testing.T) {
+	if err != nil {
+		t.Errorf("Failed to create http request with error %v", err)
+	}
+	exporter.AddHeaders(req)
+
+	// Check that all the headers are there.
+	for name, field := range testConfig.Headers {
+		// Headers are case-insensitive; Viper converts all keys to lower-case.
+		lowercaseName := strings.ToLower(name)
+		if req.Header.Get(lowercaseName) != field {
+			t.Errorf("Failed to add header: '%v' from Config.Headers", name)
+		}
+	}
+	if req.Header.Get("Content-Encoding") != "snappy" {
+		t.Errorf("Failed to add required header 'Content-Encoding'")
+	}
+	if req.Header.Get("Content-Type") != "application/x-protobuf" {
+		t.Errorf("Failed to add required header 'Content-Encoding'")
+	}
+}
+
+// TestBuildRequest tests whether a http request is a POST request, has the correct body, and has
+// the correct headers.
+// Note: this could be moved to a `cortex_internal_test.go` file as it doesn't need to be exported.
+func TestBuildRequest(t *testing.T) {
+	// Make fake exporter and message for testing.
+	var testMessage = []byte(`Test Message!`)
+	exporter := cortex.Exporter{ValidConfig}
+
+	// Create the http request.
+	req, err := exporter.BuildRequest(testMessage)
+	if err != nil {
+		t.Fatalf("Failed to build request with error %v", err)
+	}
+
+	// Verify the http method, url, and body.
+	if req.Method != http.MethodPost {
+		t.Errorf("Request is of method %v, wanted POST", req.Method)
+	}
+	if req.URL.String() != ValidConfig.Endpoint {
+		t.Errorf("Request has endpoint %v, wanted %v", req.URL, ValidConfig.Endpoint)
+	}
+	reqMessage, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		t.Errorf("Failed to read request body with error %v", err)
+	}
+	if !cmp.Equal(reqMessage, testMessage) {
+		t.Errorf("Request body has message %v, wanted %v", reqMessage, testMessage)
+	}
+
+	// Verify headers.
+	for name, field := range exporter.Config.Headers {
+		// Headers are case-insensitive; Viper converts all keys to lower-case.
+		lowercaseName := strings.ToLower(name)
+		if req.Header.Get(lowercaseName) != field {
+			t.Errorf("Failed to add header: '%v' from Config.Headers", name)
+		}
+	}
+	if req.Header.Get("Content-Encoding") != "snappy" {
+		t.Errorf("Failed to add required header 'Content-Encoding'")
+	}
+	if req.Header.Get("Content-Type") != "application/x-protobuf" {
+		t.Errorf("Failed to add required header 'Content-Encoding'")
+	}
+}
+
+// TestBuildMessage tests whether BuildMessage successfully returns a Snappy-compressed protobuf
+// message.
+// Note: Not too sure how to test this function.
+func TestBuildMessage(t *testing.T) {
+	exporter := cortex.Exporter{ValidConfig}
+	timeseries := []*prompb.TimeSeries{}
+
+	// BuildMessage simply calls protobuf.Marshal() and snappy.Encode(). BuildMessage returns the
+	// error returned by these two functions, which have their own tests in their respective
+	// packages. As long as no error is returned, the function should work as expected.
+	_, err := exporter.BuildMessage(timeseries)
```

Comment:
This unit test I feel should be integrated in the actual function. The normal functions can have some error handling, and this is simply checking if a package function doesn't check for the error. adding the err != nil to the actual BuildMessage is fine... unless you're trying to up test coverage?

Reply:
I'm trying to maintain test coverage yes